insights/client/connection.py

```diff
@@ -752,14 +754,16 @@ def _fetch_system_by_machine_id(self):
             if (self.handle_fail_rcs(res)):
                 return None
             res_json = json.loads(res.content)
+            res_json['insights_id'] = machine_id
         except ValueError as e:
             logger.error(e)
             logger.error('Could not parse response body.')
             return None
-        if res_json['total'] == 0:
-            logger.debug('No hosts found with machine ID: %s', machine_id)
+        if res.status_code == 404:
+            logger.debug(res.json()['detail'])
```

Comment:
Are we sure the response will always have `'detail'` key? The [official documentation](https://console.redhat.com/docs/api/inventory/v1#operations-hosts-api\.host\.get_host_exists) doesn't contain any schema for 404 response, I'd like to wrap this in `try/except` or otherwise ensure that we don't error out if the response is empty or doesn't contain that key.

Reply:
I would rather have there original debug message saying `No hosts found with machine ID: ...` without using `try/expect` block

Reply:
It is valuable to have the response in the log file. We can try to log the content and then always say we didn't find the host there, no matter whether we `except`ed or not.